### PR TITLE
Use resizable panel

### PR DESCRIPTION
**Pull request by rasolca**
The resizable panel has been used:
- to refactor the triangular solver
- and to simplify genToStd

Note
- [x] Depends on #391.

### CODE REVIEW

**Comment by rasolca** [comment]
bors try

File: include/dlaf/matrix/panel.h [hunk 1]
```diff
@@ -190,18 +195,68 @@ struct Panel<axis, const T, D> {
...
+    DLAF_ASSERT(width > 0, width);
+    DLAF_ASSERT(width <= parentDistribution().blockSize().cols(), width,
+                parentDistribution().blockSize().cols());
+
+    dim_ = width;
+  }
+
+  /// Set the height of the row panel.
+  ///
+  /// By default the  oghtf the panel is parentDistribution().block_size().rows().
```
Comment (Sely85) [inline include/dlaf/matrix/panel.h:216]:
```suggestion
  /// By default the height the panel is parentDistribution().block_size().rows().
```
?

Comment (rasolca) [reply include/dlaf/matrix/panel.h:216]: Fixed by https://github.com/eth-cscs/DLA-Future/pull/391/commits/c617901b9dd088835bb84e8599affc70e098b1ab in #391

**Comment by rasolca** [comment]
bors try

File: include/dlaf/matrix/panel.h (hunk 1 above)
Comment (msimberg) [reply include/dlaf/matrix/panel.h:216]: Needs a rebase?

File: include/dlaf/matrix/panel.h [hunk 2]
```diff
@@ -320,14 +373,21 @@ struct Panel : public Panel<axis, const T, device> {
   ///
   /// @pre index must point to a tile which is internally managed by the panel
   hpx::future<TileType> operator()(const LocalTileIndex& index) {
+    // Note assertion on index done by linearIndex method.
```
Comment (msimberg) [inline include/dlaf/matrix/panel.h:376]: Is it better to just put an assertion here as well? The comment and an assertion are both one line.

**Review by msimberg — APPROVED** [review]
Just a couple of minor non-blocking comments, so approving.

File: include/dlaf/matrix/panel.h (hunk 2 above)
Comment (rasolca) [reply include/dlaf/matrix/panel.h:372]: I would avoid to duplicate assertions as it makes harder to maintain the code.
Note: I fixed the assertion in `linearIndex` as it was not complete => 1 change against 4 changes (if duplicated).

Anyway... assertion was two lines (getting idx is needed as well), and now is 3 lines...

**Review by Sely85 — APPROVED** [review]
LGTM

**Review by albestro — APPROVED** [review]
LGTM

**Comment by rasolca** [comment]
bors try

**Pull request merged**